app/controllers/concerns/hyrax/collections_controller_behavior.rb

```diff
@@ -103,6 +103,8 @@ def create
       @collection.attributes = collection_params.except(:members)
       @collection.apply_depositor_metadata(current_user.user_key)
       add_members_to_collection unless batch.empty?
+      @collection.visibility = Hydra::AccessControls::AccessRight::VISIBILITY_TEXT_VALUE_PRIVATE
```

Comment:
I took a look at this and can't find where the default visibility is set.  But I did confirm that it comes into the new form with visibility='restricted' (aka private) and it always leaves the form that one because the Discovery tab is not visible until the Collection is created.

I do think that after update should force the visibility to private if discoverable? false